app/storage/metadata_parser.py

```diff
@@ -20,74 +20,45 @@ def id_generator():
     return str(uuid.uuid4())
 
 
-class MetadataField(object):
-    def __init__(self, mandatory=True, validator=None, generator=None):
-        # the function to convert the value from the jwt into the required data type
-        self._validator = validator
-        # the function to execute if the value should be auto-generated
-        self._generator = generator
-        # flag to indicate if the value must exist in the jwt token
-        self.mandatory = mandatory
-
-    def validate(self, original_value):
-        if self._validator:
-            # The parser methods throw exceptions on incorrect data
-            self._validator(original_value)
-
-    def generate(self):
-        if self._generator:
-            return self._generator()
-
-        return None
-
-
-metadata_fields = {
-    'user_id': MetadataField(),
-    'ru_ref': MetadataField(),
-    'ru_name': MetadataField(),
-    'eq_id': MetadataField(),
-    'collection_exercise_sid': MetadataField(),
-    'period_id': MetadataField(),
-    'period_str': MetadataField(),
-    'ref_p_start_date': MetadataField(validator=iso_8601_data_parser),
-    'ref_p_end_date': MetadataField(mandatory=False, validator=iso_8601_data_parser),
-    'form_type': MetadataField(),
-    'survey_url': MetadataField(mandatory=False),
-    'return_by': MetadataField(validator=iso_8601_data_parser),
-    'trad_as': MetadataField(mandatory=False),
-    'employment_date': MetadataField(mandatory=False, validator=iso_8601_data_parser),
-    'region_code': MetadataField(mandatory=False),
-    'language_code': MetadataField(mandatory=False),
-    'tx_id': MetadataField(mandatory=False, validator=uuid_4_parser, generator=id_generator),
-    'variant_flags': MetadataField(mandatory=False),
-    'roles': MetadataField(mandatory=False),
-    'case_id': MetadataField(mandatory=False),
-    'case_ref': MetadataField(mandatory=False),
-    'account_service_url': MetadataField(mandatory=False),
-}
-
-
-def parse_metadata(metadata_to_check):
-    parsed = {}
+def parse_metadata(decrypted_token, schema_metadata):
```

Comment:
Could you add a docstring to this function to explain that it's extracting values from the claims that are defined in the schema's metadata?